home/gettingstarted/0.7/javausage.md

```diff
@@ -12,66 +12,117 @@ outof: 50
-
-      public class List implements Group {
-
-Next, we declare a `Using` class for the benchmark.
-In the Scala frontend, this is achieved with the `using` block, as described in the [simple microbenchmark example](/home/gettingstarted/0.7/simplemicrobenchmark/index.html).
-
-        public class groupBy
-        implements Using<LinkedList<Integer>, HashMap<Integer, LinkedList<Integer>>> {
-
-To override the default configuration for a `Using` class or a `Group`,
-we declare a final `config` field of the type `JContext` (more about custom per-group configurations [here](/home/gettingstarted/0.7/configuration/index.html)):
-
-          public final JContext config = JContext.create()
-            .put("exec.benchRuns", 20)
-            .put("exec.independentSamples", 1)
-            .put("exec.outliers.covMultiplier", 1.5)
-            .put("exec.outliers.suspectPercent", 40);
-
-Next, we declare a generator for the `groupBy` class (see more about generators in the [generator section](/home/gettingstarted/0.7/generators/index.html)):
-
-          public JavaGenerator<LinkedList<Integer>> generator() {
-            JavaGenerator<Integer> sizes = new SingleGen("size", 5000000);
-            return new CollectionGenerators(sizes).lists();
+In the Scala frontend, we encode the groups using nested closures.
+In the Java frontend, we encode them by placing the `benchmark` annotation 
+that argument denotes the group.
+We now declare benchmark snippet stub in a `list` group:
+
+      @benchmark("list")
+      public void histogram() {
+      
+      }
+      
+Next, we want a custom context:
+
+      public final Context config = new ContextBuilder()
+          .put("exec.benchRuns", 20)
+          .put("exec.independentSamples", 1)
+          .put("exec.outliers.covMultiplier", 1.5)
+          .put("exec.outliers.suspectPercent", 40)
+          .build();
+          
+We want it not only for our histogram method 
+but for all benchmark snippets in the `list` group:
+    
+    @scopes({
+      @scopeCtx(scope = "list", context = "config")
+    })
+    public class MyHistogramTest extends JBench.OfflineReport {
+
+Next, we declare a generator for our snippet (see more about generators in the [generator section](/home/gettingstarted/0.7/generators/index.html)).
+
+      public final JGen<Integer> sizes = JGen.intValues("size", 500000);
+    
+      public final JGen<Tuple2<Integer, LinkedList<Integer>>> lists = sizes.zip(sizes.map(
+          new Fun1<Integer, LinkedList<Integer>>() {
+            public LinkedList<Integer> apply(Integer v) {
+              return new LinkedList<Integer>();
+            }
           }
+      ));
+      
+      @gen("lists")
+      @benchmark("list")
+      public void histogram(Tuple2<Integer, LinkedList<Integer>> v) {
+        
+      }
 
-Finally, we declare the code snippet for the test by implementing the `snippet` method:
+Now we initialize our list with random data in the setup method:
```

Comment:
Now, assume that the list needs to be initialized with random data before the benchmark begins.
    We will declare our own custom setup method named `listSetup` that does this initialization.
    Then, to ensure that `listSetup` is invoked before the benchmark,
    we need to add the `@setup` annotation to the `histogram` method: